File: src/dataset/typescript/isbn-verifier_exercism-wackerow/gemini/GeminiFlash_isbn-verifier_v2.ts
export default class ISBN {
  constructor(private _isbn: string) {}

  public isValid(): boolean {
    const sanitized = this._isbn.replace(/-/g, '').toLowerCase().trim();

    if (sanitized.length !== 10) {
      return false;
    }

    let sum = 0;
    for (let i = 0; i < 9; i++) {
      const digit = parseInt(sanitized[i], 10);
      if (isNaN(digit)) {
        return false;
      }
      sum += digit * (10 - i);
    }

    const lastChar = sanitized[9];
    let lastDigit: number;

    if (lastChar === 'x') {
      lastDigit = 10;
    } else {
      lastDigit = parseInt(lastChar, 10);
      if (isNaN(lastDigit)) {
        return false;
      }
    }

    sum += lastDigit;

    return sum % 11 === 0;
  }
}

File: src/dataset/python/isbn-verifier_exercism-fruit-in/gemini/GeminiFlash_isbn_verifier_v2.py
def is_valid(isbn):
    isbn = isbn.replace('-', '')
    if len(isbn) != 10:
        return False

    total = 0
    for i, digit in enumerate(isbn):
        if digit.isdigit():
            total += int(digit) * (10 - i)
        elif digit == 'X' and i == 9:
            total += 10
        else:
            return False

    return total % 11 == 0